File: frontend-web/src/lib/elapsedRealtime.ts
/** Segundos inteiros decorridos entre dois instantes (mínimo 0). */
export function elapsedWholeSeconds(entry: Date, end: Date): number {
  const sec = Math.floor((end.getTime() - entry.getTime()) / 1000)
  return Math.max(0, sec)
}

/** Texto legível pt-BR — sempre horas, minutos e segundos (contador ao vivo). */
export function formatElapsedPtBr(totalSeconds: number): string {
  const s = Math.max(0, Math.floor(totalSeconds))
  const h = Math.floor(s / 3600)
  const m = Math.floor((s % 3600) / 60)
  const sec = s % 60
  return `${h} h ${m} min ${sec} s`
}

File: android/app/src/main/java/com/estacionamento/parking/util/ElapsedRealtime.kt
package com.estacionamento.parking.util

import java.time.Instant
import java.time.temporal.ChronoUnit
import kotlin.math.max

fun elapsedWholeSeconds(entry: Instant, end: Instant): Long =
    max(0, ChronoUnit.SECONDS.between(entry, end))

/** Texto pt-BR — sempre horas, minutos e segundos (contador ao vivo). */
fun formatElapsedPtBr(totalSeconds: Long): String {
    val s = max(0, totalSeconds)
    val h = s / 3600
    val m = (s % 3600) / 60
    val sec = s % 60
    return "${h} h ${m} min ${sec} s"
}
